solr/core/src/java/org/apache/solr/handler/admin/ZookeeperStatusHandler.java

```diff
@@ -75,27 +76,18 @@ public Category getCategory() {
 
   @Override
   public void handleRequestBody(SolrQueryRequest req, SolrQueryResponse rsp) throws Exception {
-    NamedList<Object> values = rsp.getValues();
-    if (cores.isZooKeeperAware()) {
-      String zkHost = cores.getZkController().getZkServerAddress();
-      ZkDynamicConfig dynConfig = null;
-      try {
-        SolrZkClient zkClient = cores.getZkController().getZkClient();
-        dynConfig = ZkDynamicConfig.parseLines(zkClient.getConfig());
-      } catch (SolrException e) {
-        if (!(e.getCause() instanceof KeeperException)) {
-          throw e;
-        }
-        if (log.isWarnEnabled()) {
-          log.warn("{} - Continuing with static connection string", e.toString());
-        }
-      }
-      values.add("zkStatus", getZkStatus(zkHost, dynConfig));
-    } else {
-      throw new SolrException(
-          SolrException.ErrorCode.BAD_REQUEST,
-          "The Zookeeper status API is only available in Cloud mode");
-    }
+    final ZookeeperAPI zookeeperAPI = new ZookeeperAPI(cores, req, rsp);
+    V2ApiUtils.squashIntoNamedList(rsp.getValues(), zookeeperAPI.getStatus());
+  }
+
+  @Override
+  public Boolean registerV2() {
+    return true;
+  }
+
+  @Override
+  public Collection<Class<? extends JerseyResource>> getJerseyResources() {
+    return List.of(ZookeeperAPI.class);
```

Comment:
[Q] Hmm, `ZookeeperAPI` gets associated with (and registered by) both ZooKeeperInfoHandler and ZooKeeperStatusHandler...I wonder how Jersey handles that double-registration.  Just leaving this as a reminder to myself to debug this a bit.